scripts.py

```diff
@@ -1225,11 +1219,141 @@ def do_run(self):
+        editions.cover_id AS editions_cover_id, editions.cover_full_url AS editions_cover_full_url, editions.cover_thumbnail_url AS editions_cover_thumbnail_url, 
+        editions.open_access_download_url AS editions_open_access_download_url, editions.simple_opds_entry AS editions_simple_opds_entry, 
+        editions.extra AS editions_extra \n
+        FROM editions, identifiers \n
+        WHERE editions.primary_identifier_id IN (:primary_identifier_id_1) AND identifiers.type = :type_1'
+        '''
+
+        if log:
+            log.info(
+                "Processing %d editions.", query.count()
+            )
+        return query.order_by(Edition.id)
+
+
+    def run(self, batch_size=10):
+        param_args = self.parse_command_line(self._db)
+        
+        #if param_args.identifiers:
+            # we're asked about a specific set of work contributors
+            #identifier_ids = [x.id for x in param_args.identifiers]
+
+        self.query = self.make_query(
+            self._db, param_args.identifier_type, param_args.identifiers, self.log
+        )
+
+        editions = True
+        offset = 0
+        while editions:
+            editions = self.query.offset(offset).limit(batch_size).all()
+            for edition in editions:
+                if edition.contributions:
+                    for contribution in edition.contributions:
+                        self.process_contribution(self._db, contribution, edition)
+            offset += batch_size
+            self._db.commit()
+        self._db.commit()
+
+
+    @classmethod
+    def process_contribution(cls, _db, contribution, edition):
+        if not contribution or not edition:
+            return
+
+        contributor = contribution.contributor
+
+        identifier = edition.primary_identifier
+
+        computed_sort_name_local = display_name_to_sort_name(contributor.display_name, advanced=True)
+        success = cls.check_sort_name(computed_sort_name_local, contributor, edition, DataSource.INTERNAL_PROCESSING, CoverageRecord.REPAIR_SORT_NAME_OPERATION)
+        if not success:
+            output = "Contributor[%s]: contributor_sort_name=%s, contributor_display_name=%s: INTERNAL MISMATCH" % (contributor.id, contributor.sort_name, contributor.display_name)
+            print output.encode("utf8")
+
+
+    @classmethod
+    def check_sort_name(cls, computed_sort_name, contributor, edition, data_source, operation):
+        success = True
+        if computed_sort_name.strip().lower() != contributor.sort_name.strip().lower():
+            record, is_new = CoverageRecord.add_for(
+                edition=edition, data_source=DataSource.VIAF, operation=CoverageRecord.REPAIR_SORT_NAME_OPERATION,
+                status=CoverageRecord.TRANSIENT_FAILURE
+            )
+            # NOTE:  not the best way -- storing id info in the message field, but 
+            # perhaps better than making CoverageRecords cover Contributor-type objects.
```

Comment:
I'm uneasy about this for two reasons. First, as you point out, CoverageRecords aren't designed to cover Contributor-type objects. Second, this code is in core but the code for dealing with it (doing a VIAF lookup) is in metadata.

I would take this code out and let the core script do the best it can with your improved display_name_to_sort_name algorithm. In metadata, I would have the child of this script immediately do a VIAF lookup if it feels like it needs to, rather than saving the work until later.

If you really want to do a VIAF lookup from outside metadata, you can call `SimplifiedOPDSLookup.from_config()` to get an object that will let you ask metadata wrangler for the sort name of an author, using `canonicalize_author_name`. 

I think this makes sense in a repair script. Just have the metadata subclass of the script omit this step, since it _is_ the metadata wrangler and can do the VIAF lookup itself.

To summarize:

1. If a VIAF lookup needs to happen, do it immediately, don't save it for later. It'll slow down the script, but this is a repair script, we're not in a hurry here.
2. If you're not in metadata, get a `SimplifiedOPDSLookup` object and ask metadata to do the VIAF lookup.
3. If you're in metadata, do the VIAF lookup yourself.